roles/usegalaxy-eu.remove-orphan-condor-jobs/files/remove-orphan-condor-jobs

```diff
@@ -5,23 +5,28 @@
 # be removed
 source ~/.bashrc
 date
-# Get running jobs from database
-galaxy_running_jobs=(`gxadmin query queue-detail | grep running | awk '{print$3}'`)
 
+IFS=$'\n'
+# Get running jobs from the HTCondor queue
+condor_running_jobs=(`condor_q -autoformat Cmd ClusterId JobStatus | grep " 2$" | cut -d " " -f1,2 | awk '{n = split($1, arr, "/"); print arr[n-1], $2}'`)
```

Comment:
what do you all think about using -json for condor_q and jq to parse the output, it should be more stable

Reply:
Could you rebase and commit to this PR, that would help me a lot

Reply:
> Could you rebase and commit to this PR, that would help me a lot

What should I rebase, the new PR onto 347d089ab10134c53efc92c5035f929b8f133804?

Reply:
> Q: condor_q knows how long a job is in the queue, could we filter somehow by "give me all jobs that are older than 1min"?

Isn't querying for queued jobs just better? Filtering by "give me all jobs that are older than 1min" assumes that Galaxy will take less than 1 minute to update the status of a job from "queued" to "running". Which is a fair assumption, but getting also queued jobs does the same job with less asumptions, so it is simply more robust.

Let me explain myself better, we want to achieve C (do not kill jobs that show in Galaxy as queued but in HTCondor as running), and we have ways A (get both "running" and "queued" jobs with gxadmin) and B (evaluate only HTCondor jobs older than 1 minute) to achieve it, and they are not mutually exclusive. Approach B can fail in a way that approach A cannot, and this PR already applies A. If A already achieves C and is less error-prone than B, then what is the point of also using approach B?

Reply:
I also ee your point José, querying for `running` and `queued` can not do any harm, I would guess

Reply:
We should not consider queued jobs. Only running jobs that are older than X seconds.

To be safe you could also add a safeguard to the other side: That is "get all jobs from Galaxy, running AND queued". Don't kill jobs that are in the queued state in Galaxy.

Overall we want to kill jobs that have failed in Galaxy. So are neither in a running nor queued state.

Reply:
> come round

Thanks for doing so :)